### ISSUE TITLE
IndexError in _calculate_min_dose_difference_by_slice

### ISSUE THREAD

**Issue by rickardcronholm** [SimonBiggs/npgamma#18]
if to_be_checked is empty for a particular slice, the following code will throw an IndexError
```
    for current_slice in sliced:
        current_to_be_checked = np.zeros_like(to_be_checked).astype(bool)
        current_to_be_checked[[
            item[current_slice] for
            item in all_checks]] = True

```
It can be averted by returning an empty `min_dose_difference `prior to going into that loop.

**Comment by SimonBiggs** [comment]
Interesting, does that occur when the number of threads is larger than the number of slices?

**Comment by rickardcronholm** [comment]
I had just one thread. It appears to occur when there are differences in
coordinates. Specifically when one dose distribution has values well beyond
the other. I can share dataset that will reproduce the error.

Den 6 maj 2017 12:23 skrev "Simon Biggs" <notifications@github.com>:

**Comment by SimonBiggs** [comment]
Yes please. That appears to be related to something I most recently changed. So that makes sense.

I'll see if I can dive in and improve it.

**Comment by SimonBiggs** [comment]
An example dataset will be helpful.

**Comment by SimonBiggs** [comment]
Hi Richard,

Would you be able raise this as an issue on GitHub. I'm not sure if I'm
going to get any time to work on npgamma, but at least if it's an issue on
GitHub people can see it, maybe they could fix it, or at least be aware
that it is an issue.
